src/DiffEqDiffTools.jl

```diff
@@ -1,5 +1,74 @@
 module DiffEqDiffTools
 
 # package code goes here
+# testing refactoring PR
+
+immutable DiffCache{T, S}
+    du::Vector{T}
+    dual_du::Vector{S}
+end
+
+Base.@pure function DiffCache{chunk_size}(T, length, ::Type{Val{chunk_size}})
+    DiffCache(zeros(T, length), zeros(Dual{chunk_size, T}, length))
+end
+
+Base.@pure DiffCache(u::AbstractArray) = DiffCache(eltype(u),length(u),Val{ForwardDiff.pickchunksize(length(u))})
+Base.@pure DiffCache(u::AbstractArray,nlsolve) = DiffCache(eltype(u),length(u),Val{get_chunksize(nlsolve)})
+Base.@pure DiffCache{CS}(u::AbstractArray,T::Type{Val{CS}}) = DiffCache(eltype(u),length(u),T)
+
+get_du{T<:Dual}(dc::DiffCache, ::Type{T}) = dc.dual_du
```

Comment:
You're going to need to `import ForwardDiff: Dual` in order to use `Dual` like this. The same statement should be removed from OrdinaryDiffEq and StochasticDiffEq.